internal/pkg/diagnostics/diagnostics_test.go

```diff
@@ -140,6 +140,103 @@ mapping:
 	require.Empty(t, errOut.String())
 }
 
+func TestRedactWithSecretPaths(t *testing.T) {
+	tests := []struct {
+		name   string
+		input  []byte
+		expect string
+	}{{
+		name: "no secret paths",
+		input: []byte(`id: test-policy
+inputs:
+  - type: test_input
+    redactKey: secretValue
+outputs:
+  default:
+    type: elasticsearch
+    api_key: secretKey
+    redactOtherKey: secretOutputValue
+`),
+		expect: `id: test-policy
+inputs:
+    - redactKey: secretValue
+      type: test_input
+outputs:
+    default:
+        api_key: <REDACTED>
+        redactOtherKey: secretOutputValue
+        type: elasticsearch
```

Comment:
I think these would be more readable if inputs and outputs used the same indentation.